### DISCUSSION TITLE
Expose the `gray-matter` option to config

### DISCUSSION THREAD

**Discussion by Yixuan-Wang** [withastro/roadmap#187] [Proposal]
This will introduce a new option in the Markdown section of Astro config, which will be passed into the `gray-matter` frontmatter parsing engine. Additionally, the `excerpt` returned by `gray-matter` of type `string | undefined` will be exposed to the Markdown Astro component or the raw Markdown file.

This enables some new features:
- Use TOML or other frontmatter languages in the frontmatter
- Export an excerpt, which is very useful in blogging (this is provided by default in other blogging solutions like Hexo and Hugo etc.)

I have done some work on my branch at https://github.com/Yixuan-Wang/astro/tree/feat-graymatter. If this is acceptable, I'll file a PR.

**Comment by colinbate** [comment]
I've created a remark plugin which supports TOML frontmatter (including layout mapping), but it is needlessly complicated for something that could be done with support for a custom frontmatter provider.

**Reply by delucis** [reply]
TOML frontmatter is supported by default since Astro v5.2! https://astro.build/blog/astro-520/#toml-frontmatter-in-markdown
